### PR TITLE
feat: Don't repeat CI authentication docs

### PR DESCRIPTION
**Pull request by erickzanardo**
## Status

**READY**

## Description

This PR extracts the part of the docs that explain how to obtain a CI token in its own reusable file. Which then is used on both the github and codemagic guides. Removing the duplicated docs between both files.

This will make this section of the docs easier to scale as we add more content.

After this, I plan on adding to the `_ci_authentication.md` file about how the token can also be acquired from the console.

### CODE REVIEW

File: docs/ci/github.mdx [hunk 1]
```diff
@@ -54,38 +56,7 @@ Currently `setup-shorebird` only supports the latest stable version of shorebird
 
 ## Authentication
```
Comment (felangel) [inline docs/ci/github.mdx:57]: Can we include the Authentication heading as part of the reusable CIAuthentication component?

**Review by felangel — APPROVED** [review]
LGTM just left one comment about moving the heading into the component as well 👍

File: docs/ci/github.mdx (hunk 1 above)
Comment (erickzanardo) [reply docs/ci/github.mdx:57]: That was my first approach, but when doing so, the section would not show in the index 🙃

Comment (felangel) [reply docs/ci/github.mdx:57]: Ah that’s annoying haha

**Pull request merged**